lib/src/back_end/code_writer.dart

```diff
@@ -101,31 +86,16 @@ class CodeWriter {
   /// solution if the line ends up overflowing.
   final List<Piece> _currentUnsolvedPieces = [];
 
-  /// The options for the current innermost piece being formatted.
-  _PieceOptions get _options => _pieceOptions.last;
+  CodeWriter(this._pageWidth, this._solution);
 
-  /// The offset in the formatted code where the selection starts.
+  /// Called after this writer has finished formatting the entire piece tree.
```

Comment:
[nit] Prefer to describe a method, at least in the header, by what it does instead of when it is called.

Reply:
Just removed the first sentence.